### DISCUSSION TITLE
T12-958 v2 few questions

### DISCUSSION THREAD

**Discussion by mako777** [deividAlfa/stm32_soldering_iron_controller#138] [General]
Hi, I have this station, bought it a few months ago and recently flashed your firmware and can confirm it is working great. My MCU was also not read protected so I was able to dump original firmware, so if you want I can share it. I've also compared my hex with the file you have in this repo and it is mostly different (only few blocks are the same, maybe different sw version).

In recent release you wrote "Dedicated Stand input where supported (T12-958 v2 has this)." can you share some more info about this ?

Do you have plans to make some sort of screen saver, like in the original firmware after a minute or so it switches to "moving display" to prevent OLED wear ?

I can help with some testing if there is a need to.
Regards.

**Comment by deividAlfa** [comment]
Oled doesn't burn like crt or plasma, simply each pixel has a lifespan, becoming dimmer as it's used up, this fw uses dimming, which can be enabled even in working state, also can shut it down completely when the tip fully cold and safe to touch, which is better than moving a picture at full brightness (Will wear the oled down anyways).

About the fw, yeah, there's never too many backups! 🙂

**Comment by mako777** [comment]
Yes, I know about the dimming and how OLED wears, I was simply thinking about making even more wear protection, because static image (mostly static ui elements), even in lowest brightness, will slowly wear the same pixels and create dark spots, but if there is some movement then it should wear move evenly.

And what about this dedicated stand input, can you share some more info, because I've gone through the docs and there is no info about this. Mostly I would like to know where I'm supposed to hook this up.

Org fw dump from my station:
[Quecoo T12-958 hw3.20 sw3.2S STM32F103 OLED.bin.zip](https://github.com/user-attachments/files/17789523/Quecoo.T12-958.hw3.20.sw3.2S.STM32F103.OLED.bin.zip)

**Comment by deividAlfa** [comment]
![IMG_20241117_124110](https://github.com/user-attachments/assets/fd0da40a-0a11-4b16-b225-05680d290a44)

Leave floating for working operation, pull to ground to set sleep mode.
Normally connected to the stand and the handle makes the ground, but you can also use a switch or whatever it comes to your mind.

**Comment by mako777** [comment]
Yeah, I've connected the stand and it is working great (I had to file the stand and handle a little so it makes good contact but this is no big deal). This soldering station was pretty good from the beginning but now it's even better. Thank you for this firmware and have a good day.
